### PR TITLE
Added Clip layer for clipping of outputs

### PR DESCRIPTION
**Pull request by patyork**
**DON'T MERGE - LET ME KNOW IF MERGE #36 IS USED - Either way, there is an additional commit that will need to be applied.**

This provides the ability to apply a clipping to a Layer. Clipping is sometimes used to defend against output saturation and vanishing gradient.

EX:
```python
model = Sequential()
model.add(Dense(2,1, activation='sigmoid', init='example_normal')) # this example_normal is an additional PR
model.add(Clip(high=.32, low=.20))


sgd = SGD(lr=0.1, decay=1e-6, momentum=0.9, nesterov=True)
model.compile(loss='mse', optimizer='sgd')

A = model._predict(np.asarray([[1,1],[2,2]]))
print A
```
Yields
```
[[ 0.31999999]
 [ 0.2       ]]
```

### CODE REVIEW

**Comment by alito** [comment]
So what's the problem with this patch? Are clip layers not useful?
